File: Keeper/Models/Keep.cs


namespace Keeper.Models;

public class Keep : RepoItem<int>
{
    public string CreatorId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string Img { get; set; }
    public int Views { get; set; }
    public Profile Creator { get; set; }
    public int kept { get; set; }
}

public class TrueKeep : Keep
{
    public int VaultKeepId { get; set; }
}

File: Keeper.client/src/models/Keep.js
import { RepoItem } from "./RepoItem.js"

export class Keep extends RepoItem {
  constructor(data) {
    super(data)
    this.creatorId = data.creatorId
    this.creator = data.creator
    this.description = data.description
    this.img = data.img
    this.views = data.views
    this.kept = data.kept
    this.name = data.name
    this.vaultKeepId = data.vaultKeepId
  }
}